src/ess/polarization/base.py

```diff
@@ -379,41 +368,92 @@ def intensity(wavelength: sc.Variable, opacity0: sc.Variable) -> sc.Variable:
         p0={'opacity0': sc.scalar(1.0, unit='1/nm')},
     )
-    return He3OpacityFunction[Cell](OpacityFunction(popt['opacity0'].data))
+    return He3OpacityFunction[Cell](popt['opacity0'].data)
+
+
+class He3PolarizationFunction(Generic[Cell]):
+    """Time-dependent polarization function for a given cell."""
+
+    def __init__(self, C: sc.Variable, T1: sc.Variable):
+        self._C = C
+        self._T1 = T1
+
+    @property
+    def C(self) -> sc.Variable:
+        return self._C
+
+    @property
+    def T1(self) -> sc.Variable:
+        return self._T1
+
+    def __call__(self, time: sc.Variable) -> sc.Variable:
+        return self.C * sc.exp(-time / self.T1)
+
+
+@dataclass
+class He3TransmissionFunction(Generic[Cell]):
+    """Wavelength- and time-dependent transmission for a given cell."""
+
+    opacity_function: He3OpacityFunction[Cell]
+    polarization_function: He3PolarizationFunction[Cell]
+    transmission_empty_glass: He3TransmissionEmptyGlass[Cell]
+
+    def __call__(self, time: sc.Variable, wavelength: sc.Variable) -> sc.Variable:
+        opacity = self.opacity_function(wavelength)
+        polarization = self.polarization_function(time)
+        return (
+            self.transmission_empty_glass
+            * sc.exp(-opacity)
+            * sc.cosh(opacity * polarization)
+        )
 
 
 def he3_polarization(
     direct_beam_no_cell: DirectBeamNoCell,
     direct_beam_polarized: He3DirectBeam[Cell, Polarized],
-    opacity: He3OpacityFunction[Cell],
-    filling_time: He3FillingTime[Cell],
+    opacity_function: He3OpacityFunction[Cell],
     transmission_empty_glass: He3TransmissionEmptyGlass[Cell],
-) -> He3Polarization[Cell]:
+) -> He3PolarizationFunction[Cell]:
     """
-    Fit time- and wavelength-dependent equation and return the fit param P(t).
+    Fit time- and wavelength-dependent equation and return the fitted P(t).
 
     DB_pol/DB = T_E * cosh(O(lambda)*P(t))*exp(-O(lambda))
     """
-    # Each time bin corresponds to a direct beam measurement. Take the mean for each
-    # but keep the time binning.
-    # time_up = direct_beam_up.bins.coords['time'].bins.mean()
-    # time_down = direct_beam_down.bins.coords['time'].bins.mean()
-    # results dims: spin state, wavelength, time
-    raise NotImplementedError()
-    return He3Polarization[Cell](1)
+
+    def direct_beam_ratio(
+        wavelength: sc.Variable, time: sc.Variable, C: sc.Variable, T1: sc.Variable
+    ) -> sc.Variable:
+        polarization_function = He3PolarizationFunction[Cell](C=C, T1=T1)
+        return He3TransmissionFunction[Cell](
+            opacity_function=opacity_function,
+            polarization_function=polarization_function,
+            transmission_empty_glass=transmission_empty_glass,
+        )(time, wavelength)
+
+    popt, _ = sc.curve_fit(
+        ['wavelength', 'time'],
+        direct_beam_ratio,
+        direct_beam_no_cell / direct_beam_polarized,
```

Comment:
It should be vice versa: direct_beam_polarized / direct_beam_no_cell

Reply:
Should be fixed, see update.

Reply:
Thanks!